File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter09/MoreXMLFormatter/CustomerLibrary/Customer.cs
using System;

namespace CustomerLibrary
{
   public class Customer
   {
      public string Name;         // Public field
      private string mCreditCard; // Private field
      
      private string mEmail;  // Private field with public property
      public string Email
      { 
         get {return mEmail;}
         set {mEmail = value;}
      }

      public Customer(string name, string email, string ccNum)
      { 
         Name = name; mEmail = email; mCreditCard = ccNum;
      }  
      
      // Required for serialization    
      public Customer(){}
   }
}

File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter09/BinaryFormatter/CustomerLibrary/Customer.vb
<Serializable()> _
Public Class Customer

   Public Name As String         'Public field
   Private mCreditCard As String 'Private field     
   Private mEmail As String      'Private field with public property

   Public Property Email() As String
      Get
         Return mEmail
      End Get
      Set(value as String)
         mEmail = value
      End Set         
   End Property

   Public Sub New(name As String, email As String, ccNum As String)
      Me.Name = name : mEmail = email : mCreditCard = ccNum
   End Sub 

   'Required for serialization   
   Public Sub New()
   End Sub
   
End Class
